napari/_vispy/utils.py

```diff
@@ -16,17 +18,17 @@
     Tracks: VispyTracksLayer,
 }
 
-if config.async_loading:
+if config.async_octree:
     from ..layers.image.experimental.octree_image import OctreeImage
     from .experimental.vispy_tiled_image_layer import VispyTiledImageLayer
 
-    # Put OctreeImage in front so we hit that before plain Image
-    original = layer_to_visual.copy()
-    layer_to_visual = {OctreeImage: VispyTiledImageLayer}
-    layer_to_visual.update(original)
+    # Insert OctreeImage in front so it gets picked over plain Image.
+    new_order = {OctreeImage: VispyTiledImageLayer}
+    new_order.update(layer_to_visual)
+    layer_to_visual = new_order
 
 
-def create_vispy_visual(layer):
+def create_vispy_visual(layer: Layer, camera: Camera) -> VispyBaseLayer:
```

Comment:
I'm not sure about this change. I don't necessarily have a huge objection to it, but I probably would have done this here https://github.com/napari/napari/blob/master/napari/_qt/qt_viewer.py#L315-L317 by adding a
```python
vispy_layer = create_vispy_visual(layer)
self.viewer.camera.events.center.connect(vispy_layer._on_camera_move)
```
I think this would achieve the same outcome but without changing this function

Curious about your thoughts on the relative merits of either approach?

Reply:
Yeah I changed it to that, much better. It was just an artifact of how I arrived at this. First I was passing the camera INTO the VispyLayer. Then I changed to connecting the event and didn't think to pull it out. So this is better.

Reply:
ok, nice. thanks!